src/bindings/controller_lua.c

```diff
@@ -0,0 +1,101 @@
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+#include <string.h>
+#include "controller_lua.h"
+#include "../input/controller.h"
+#include "../debug/debug.h"
+
+static Controller *CheckController(lua_State *state);
+static int LUA_CONTROLLER_AUX_TABLE_REF = 0;
+
+static Controller *CheckController(lua_State *state)
+{
+
+    Controller **controller = (Controller **)luaL_checkudata(state, 1, "Lua.Controller");
+    luaL_argcheck(state, controller != NULL, 1, "Controller expected");
+    return *controller;
+}
+
+
+static int LuaCheckIfButtonPressed(lua_State *state)
+{
+    Controller *controller = CheckController(state);
+    int button = luaL_checknumber(state, 2);
+    lua_pushboolean(state, IsControllerButtonPressed(controller, button));
+    return 1;
+}
+
+static int LuaCheckIfButtonHeld(lua_State *state)
+{
+    Controller *controller = CheckController(state);
+    int button = luaL_checknumber(state, 2);
+    lua_pushboolean(state, IsControllerButtonHeld(controller, button));
+    return 1;
+}
+
+static int LuaCheckIfButtonReleased(lua_State *state)
+{
+    Controller *controller = CheckController(state);
+    int button = luaL_checknumber(state, 2);
+    lua_pushboolean(state, IsControllerButtonReleased(controller, button));
+    return 1;
+}
+
+static int LuaCreateController(lua_State *state)
+{
+    Controller **controller_ptr_ptr = (Controller **)lua_newuserdata(state, sizeof(Controller *));
```

Comment:
Probably should not use a ptr_ptr here, if we want lua to handle the lifetime of the controller.

Reply:
We could actually add in a _gc function potentially.

Reply:
too complex, later maybe